### PR TITLE
Fix gradient-checkpointing option is ignored in SDXL+LoRA training. (#6388)

### LINKED ISSUE #6388: gradient-checkpointing ignored in SDXL+LoRA training
There is `--gradient-checkpointing` option, but never handled. Is this intended?

https://github.com/huggingface/diffusers/blob/9f283b01d20aa8d9e5251dbaad19432bfb2ffe11/examples/text_to_image/train_text_to_image_lora_sdxl.py#L302-306

### PR DESCRIPTION
**Pull request by 2510**
# What does this PR do?

Fixes a problem in SDXL LoRA training script that ignores gradient checkpointing option.

Fixes #6388

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
  * No, this PR fixes only a bug in implementation. No docs affected.
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
  * https://github.com/huggingface/diffusers/issues/6388
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
  * No docs needs update.
- [ ] Did you write any new necessary tests?
  * I don't have good way to test this bugfix.

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by 2510** [comment]
@sayakpaul Here is a PR for #6388.
I found same problem for SD1-LoRA and fixed in the same way.

File: examples/text_to_image/train_text_to_image_lora_sdxl.py [hunk 1]
```diff
@@ -706,6 +706,9 @@ def load_model_hook(models, input_dir):
     accelerator.register_save_state_pre_hook(save_model_hook)
     accelerator.register_load_state_pre_hook(load_model_hook)
 
+    if args.gradient_checkpointing:
+        unet.enable_gradient_checkpointing()
```
Comment (sayakpaul) [inline examples/text_to_image/train_text_to_image_lora_sdxl.py:710]: The text encoder can be trainable in this script too:
https://github.com/huggingface/diffusers/blob/bf725e044e0b2fec29c074510c2d12d5ef66d2d1/examples/text_to_image/train_text_to_image_lora_sdxl.py#L621

This is how it's done:
https://github.com/huggingface/diffusers/blob/bf725e044e0b2fec29c074510c2d12d5ef66d2d1/examples/dreambooth/train_dreambooth_lora_sdxl.py#L976

**Review by sayakpaul — COMMENTED** [review]
Left a comment. Thank you.

**Comment by 2510** [comment]
@sayakpaul Thanks for your review. Fixed that in [8d7feb6](https://github.com/huggingface/diffusers/pull/6402/commits/8d7feb69cacfccaaa465ca70874921602f7b62b3).

**Review by sayakpaul — APPROVED** [review]

**Comment by sayakpaul** [comment]
Thanks for your contributions.
